apps/next/src/pages/AccountManagement/components/ImportKeystoreFile/hooks/useImportKeystoreFile.ts

```diff
@@ -1,38 +1,38 @@
-import { toast } from '@avalabs/k2-alpine';
-import { useAnalyticsContext, useKeystoreFileImport } from '@core/ui';
+import { useKeystoreFileImport } from '@core/ui';
 import { useCallback, useState } from 'react';
-import { useTranslation } from 'react-i18next';
-import { useHistory } from 'react-router-dom';
 
-export const useImportKeystoreFile = () => {
-  const { capture } = useAnalyticsContext();
-  const { t } = useTranslation();
-  const { replace } = useHistory();
+export type Callbacks = {
+  onStarted?: VoidFunction;
+  onSuccess?: VoidFunction;
+  onFailure?: VoidFunction;
+};
+
+export const useImportKeystoreFile = ({
+  onStarted,
+  onSuccess,
+  onFailure,
+}: Callbacks) => {
   const { importKeystoreFile } = useKeystoreFileImport();
 
   const [isImporting, setIsImporting] = useState(false);
 
   const importFile = useCallback(
-    async (file: File, password: string, onError: (error: unknown) => void) => {
+    async (file: File, password: string) => {
       if (isImporting) {
         return;
       }
       setIsImporting(true);
       try {
-        capture('KeystoreFileImportStarted');
+        onStarted?.();
         await importKeystoreFile(file, password);
-
-        capture('KeystoreFileImportSuccess');
-        toast.success(t('Successfully imported the keystore file.'));
-        replace('/account-management');
-      } catch (err: unknown) {
-        capture('KeystoreFileImportFailure');
-        onError(err);
+        onSuccess?.();
+      } catch {
+        onFailure?.();
       } finally {
         setIsImporting(false);
       }
     },
-    [isImporting, capture, importKeystoreFile, replace, t],
+    [isImporting, importKeystoreFile, onStarted, onSuccess, onFailure],
```

Comment:
If we add the callbacks to `useKeystoreFileImport()` hook (the one being used in line `15` here), this new hook (`useImportKeystoreFile`) will no longer be needed.

Unless I'm misunderstanding something :)